raphtory/src/db/graph/views/filter/model/exploded_edge_filter.rs

```diff
@@ -0,0 +1,851 @@
+            CompositeExplodedEdgeFilter::And(left, right) => write!(f, "({} AND {})", left, right),
+            CompositeExplodedEdgeFilter::Or(left, right) => write!(f, "({} OR {})", left, right),
+            CompositeExplodedEdgeFilter::Not(filter) => write!(f, "(NOT {})", filter),
+        }
+    }
+}
+
+impl CreateFilter for CompositeExplodedEdgeFilter {
+    type EntityFiltered<'graph, G: GraphViewOps<'graph>> = Arc<dyn BoxableGraphView + 'graph>;
+    type NodeFilter<'graph, G>
+        = NotANodeFilter
+    where
+        Self: 'graph,
+        G: GraphView + 'graph;
+
+    fn create_filter<'graph, G: GraphViewOps<'graph>>(
+        self,
+        graph: G,
+    ) -> Result<Self::EntityFiltered<'graph, G>, GraphError> {
+        match self {
+            Self::Src(filter) => {
+                let filtered_graph = filter.create_filter(graph.clone())?;
+                Ok(Arc::new(EdgeNodeFilteredGraph::new(
+                    graph,
+                    Endpoint::Src,
+                    filtered_graph,
+                )))
+            }
+            Self::Dst(filter) => {
+                let filtered_graph = filter.create_filter(graph.clone())?;
+                Ok(Arc::new(EdgeNodeFilteredGraph::new(
+                    graph,
+                    Endpoint::Dst,
+                    filtered_graph,
+                )))
+            }
+            Self::Property(p) => Ok(Arc::new(p.create_filter(graph)?)),
+            Self::PropertyWindowed(pw) => Ok(Arc::new(pw.create_filter(graph)?)),
+            Self::And(l, r) => {
+                let (l, r) = (*l, *r); // move out, no clone
+                Ok(Arc::new(
+                    AndFilter { left: l, right: r }.create_filter(graph)?,
+                ))
+            }
+            Self::Or(l, r) => {
+                let (l, r) = (*l, *r);
+                Ok(Arc::new(
+                    OrFilter { left: l, right: r }.create_filter(graph)?,
+                ))
+            }
+            Self::Not(f) => {
+                let base = *f;
+                Ok(Arc::new(NotFilter(base).create_filter(graph)?))
+            }
+        }
+    }
+
+    fn create_node_filter<'graph, G: GraphView + 'graph>(
+        self,
+        _graph: G,
+    ) -> Result<Self::NodeFilter<'graph, G>, GraphError> {
+        Err(GraphError::NotNodeFilter)
+    }
+}
+
+impl TryAsCompositeFilter for CompositeExplodedEdgeFilter {
+    fn try_as_composite_node_filter(&self) -> Result<CompositeNodeFilter, GraphError> {
+        Err(GraphError::NotSupported)
+    }
+
+    fn try_as_composite_edge_filter(&self) -> Result<CompositeEdgeFilter, GraphError> {
+        Err(GraphError::NotSupported)
+    }
+
+    fn try_as_composite_exploded_edge_filter(
+        &self,
+    ) -> Result<CompositeExplodedEdgeFilter, GraphError> {
+        Ok(self.clone())
+    }
+}
+
+#[derive(Clone, Debug, Copy, PartialEq, Eq)]
+pub struct ExplodedEdgeEndpoint(Endpoint);
```

Comment:
need to make sure these filters are applied correctly, they should apply to the node at the `time` of the event and in the `layer` of the event.